applications/Chat/coati/ray/detached_trainer_base.py

```diff
@@ -46,9 +46,9 @@ def __init__(self,
 
         self._debug = debug
 
-    def update_target_holder_list(self, experience_maker_holder_name_list):
-        self.target_holder_name_list = experience_maker_holder_name_list
-        self.target_holder_list = []
+    def update_target_holder_list(self):
+        if len(self.target_holder_list) > 0:
```

Comment:
Previous design made dynamic update possible.

Reply:
Why?

Reply:
It is not necessary but possible. One can start a new process and get the trainer remote reference. Then call trainer.update_target_holder_list.remote(new_list) to reschedule. Not necessary though.

Reply:
But this needs to rebuild process group to ensure the correctness of DP.

Reply:
More works to do.